### ISSUE TITLE
3.0 strange font issue

### ISSUE THREAD

**Issue by sedlund** [ogri-la/strongbox#191]
openjdk-11-jre:amd64                          11.0.8+10-0ubuntu1~20.04
Kubuntu 20.04 

Also the font color in the bottom area where it says checked XX addons and XX updates available  - cannot read white on white.

![image](https://user-images.githubusercontent.com/8109138/95683298-f571f280-0baf-11eb-8dbe-a3cc75800c38.png)

EDIT: just tried the AppImage - does the same thing.

**Comment by torkus** [comment]
that is strange! thank you very much for the feedback

**Comment by torkus** [comment]
just for reference, this is what it should look like

![Screenshot at 2020-10-12 08-41-37](https://user-images.githubusercontent.com/48790775/95691436-c8cce200-0c66-11eb-81ae-66f8a1127d63.png)

**Comment by torkus** [comment]
Using a brand new kubuntu image I couldn't replicate this via the code or the appimage.

Could you paste the contents or `fc-list` for me? It's a listing of all installed fonts. I could diff it against the fonts installed on my vm. I'm not specifying any fonts specifically, just using two font-families "sans" and "monospace". That said, it looks like you have the same fonts, they're just behaving strangely.

Could you also paste the contents of `locale` for me? The arrow next to the value in the 'source' column is a unicode glyph and it's not present for you, so it may be related to that.

**Comment by sedlund** [comment]
[fc-list.txt](https://github.com/ogri-la/strongbox/files/5366347/fc-list.txt)

LANG=en_US.UTF-8
LANGUAGE=en_US
LC_CTYPE="en_US.UTF-8"
LC_NUMERIC=en_US.UTF-8
LC_TIME=en_US.UTF-8
LC_COLLATE=C
LC_MONETARY=en_US.UTF-8
LC_MESSAGES="en_US.UTF-8"
LC_PAPER=en_US.UTF-8
LC_NAME=en_US.UTF-8
LC_ADDRESS=en_US.UTF-8
LC_TELEPHONE=en_US.UTF-8
LC_MEASUREMENT=en_US.UTF-8
LC_IDENTIFICATION=en_US.UTF-8
LC_ALL=


I used Lutris to install WoW - I believe it installs a bunch of other fonts that aren't in a vanilla Kubuntu

Thanks for looking at this!

**Comment by torkus** [comment]
I've ruled out locales and missing fonts, I don't think either of them are the answer here.

Are you using a KDE theme by chance? I've tried the three default themes and they seem ok, I'm looking at other ones now to see if they affect word spacing

**Comment by sedlund** [comment]
I installed Layan, and Arc Dark, but just went back to Breeze Dark (a default).  I tried a few the other day while it strongbox was running and didn't change anything.

Running it with strongbox --ui gui1 it brings up the old UI and it looks fine.

Is the new GUI toolkit using QT or GTK3 or something else?

**Comment by torkus** [comment]
> Is the new GUI toolkit using QT or GTK3 or something else?

it's using JavaFX, which is the replacement for Java Swing. I think under the hood it's just HTML and CSS using WebKit.

**Comment by torkus** [comment]
Could you confirm this issue is still affecting you with 3.0.0?

**Comment by sedlund** [comment]
> Could you confirm this issue is still affecting you with 3.0.0?

yes, it still shows the same.  I tried replicating my Kubuntu install with lutris in a VM as well, and the old pre-release worked fine there.  Must be some strange thing on my install.

**Comment by 911992** [comment]
Please also mind, that text-rendering in java(swing at least I know) is depends on underlying shell font shaping/rendering. Am not sure about the jfx, but surely the webview related stuffs are depends on hybrid(mostly native) webkit build, which is itself would depends on underlying font context I think.

As it's native, so any shell may have diff policies for rendering, and yes bugs would make things worse.

You may check your OS/Shell text rendering settings please, and perform a safe/default and try again.
Hope it helps

**Comment by torkus** [comment]
> yes, it still shows the same. I tried replicating my Kubuntu install with lutris in a VM as well, and the old pre-release worked fine there. Must be some strange thing on my install.

bugger. KDE and WebView are tightly coupled historically, there may be some system setting affecting it (but that's a stretch). Do you have another desktop environment installed like XFCE or Mate or Gnome? Those are all GTK and you might not want to dirty your system up installing them.

If you find some way of affecting it let me know. I'll keep this ticket open indefinitely. Any one else also affected should put their hand up.

**Comment by sedlund** [comment]
Figured it out!  I used `fc-match sans` which output 

`NotoSansCJK-Regular.ttc: "Noto Sans CJK SC" "Regular" `

I had no idea what SC was -- apparently it's the Simplified Chinese font.

Tried changing it to defaults in the KDE font manager, which changes all the dropdowns to regular 'Noto sans 10pt'  click ok, come back and still, the SC font was selected.

Found that I was missing package `fonts-noto-core`

`sudo apt install fonts-noto-core` -- then back to font configure and click defaults and ok.  

Now strongbox is looking good! 

Thanks for your help.
